src/main/java/com/fortify/fod/Main.java

```diff
@@ -35,11 +35,20 @@ public static void main(String[] args) {
             System.exit(1);
         }
 
-        if (fc.bsiToken == null && (fc.url == null && ((fc.url.contains("http://") || fc.url.contains("https://"))))) {
-            System.err.println("Please provide Valid Fortify Url Parameter ( Example : https://ams.fotify.com/");
-            System.exit(1);
+        if (fc.bsiToken == null)
+        {
+            if(fc.portalUri == null || (!(fc.portalUri.contains("http://") || !(fc.portalUri.contains("https://")))))
```

Comment:
optional: since you do it couple of times, it's best to create an helper method to check URL validness. that helper method would use Regex probably.

Reply:
Thinking about the same , also wanted to add more but wanted to see Qa urls because i want the urls to be limited to fod or fortify